### DISCUSSION TITLE
simple mesh_boolean operation produces Eigen assertion fail

### DISCUSSION THREAD

**Discussion by plut** [libigl/libigl#1866] [Q&A] [answered]
I try building a simple example of `mesh_boolean` operation (the union of two tetrahedra):

    Eigen::MatrixXd v1(4,3); v1 << 0,0,0, 2,0,0, 0,2,0, 0,0,2; 
    Eigen::MatrixXi f1(4,3); f1 << 1,3,2, 1,4,3, 1,2,4, 2,3,4; 
    Eigen::MatrixXd v2(4,3); v2 << 2,0,0, 0,0,0, 2,0,2, 2,2,0; 
    Eigen::MatrixXi f2(4,3); f2 << 1,3,2, 1,4,3, 1,2,4, 2,3,4; 
    Eigen::MatrixXd v3; 
    Eigen::MatrixXi f3; 
 
    std::cout << "matrix v1:\n" << v1 << "\nmatrix f1:" << f1 << "\n"; 
    std::cout << "matrix v2:\n" << v2 << "\nmatrix f2:" << f2 << "\n"; 
    std::cout << "calling mesh_boolean\n"; 
    igl::copyleft::cgal::mesh_boolean(v1,f1,v2,f2,igl::MeshBooleanType(0), v3, f3); 
    std::cout << "done!\n"; 

This code produces the following Eigen assertion fail:

    test: /home/jerome/src/ConstructiveGeometry/iglwrap/_deps/libigl-src/external/eigen/Eigen/src/Core/DenseCoeffsBase.h:364:
    Eigen::DenseCoeffsBase<Derived, 1>::Scalar& Eigen::DenseCoeffsBase<Derived, 1>::operator()(Eigen::Index, Eigen::Index)
    [with Derived = Eigen::Matrix<CGAL::Lazy_exact_nt<CGAL::Gmpq>, -1, 3, 0, -1, 3>;
     Eigen::DenseCoeffsBase<Derived, 1>::Scalar = CGAL::Lazy_exact_nt<CGAL::Gmpq>;
     Eigen::Index = long int]:
    Assertion `row >= 0 && row < rows() && col >= 0 && col < cols()' failed.

I am likely the one to do something wrong here, but in such a simple code, I cannot see where. Does anybody have an idea what happened? Should I have allocated the matrices `v3` and `f3` to some random given size? At least allocating them as `(0,3)` did not change anything.

**Comment by alecjacobson** [comment, marked as answer]
Looks like it might be as simple as switching your faces from starting at 1 to starting at 0.

**Reply by plut** [reply]
Oops! Indeed, I copied from Julia code and spent so much time checking that the face orientation was good... that I completely forgot about 1-indexing! Many thanks!
